core/src/main/java/hudson/tools/JDKInstaller.java

```diff
@@ -476,17 +477,41 @@ public URL locate(TaskListener log, Platform platform, CPU cpu) throws IOExcepti
                     throw new IOException("Failed to request " + m.getURI() +" exit code="+r);
 
                 if (m.getURI().getHost().equals("login.oracle.com")) {
+                    /* Login flow:
+                     * 1. /oaam_server/oamLoginPage.jsp: Form for username + password. Submit action is:
+                     * 2. /oaam_server/login.do: Returns a 302 to:
+                     * 3. /oaam_server/loginAuth.do: After 2 seconds, JS sets window.location to:
+                     * 4. /oaam_server/authJump.do: Contains a single form with hidden inputs and JS that submits the form to:
+                     * 5. /oam/server/dap/cred_submit: Returns a 302 to:
+                     * 6. https://edelivery.oracle.com/osso_login_success: Returns a 302 to the download.
```

Comment:
Does this comment seem useful? I created it to keep track of what was happening, but I'm not sure if it's worth keeping.

Reply:
LGTM